README.md

```diff
@@ -378,6 +378,25 @@ The default is defined as below.
 #define PCC_FREE(auxil, ptr) free(ptr)
 ```
 
+**`PCC_DEBUG(`**_event_**`,`**_rule_**`,`**_level_**`,`**_pos_**`,`**_buffer_**`,`**_length_**`)`**
+
+The function macro for debugging. Sometimes, especially for complex parsers, it is useful to see how exactly the parser processes the input. This macro is called on important *events* and allows to log or display current state of the parsers. The argument `rule` is a string that contains name of currently evaluated rule. Non-negative integer `level` is a specifying how deep in the rule hierarchy the parser currently is. Argument `pos` holds position from the start of current context in bytes. In case of `event == DBG_MATCH` argument `buffer` holds matched input and `length` is its size. for other events `buffer` and `length` describe part of currently loaded input, that is used to evaluate rule.
+
+There are currently three supported events:
+ - `DBG_EVALUATE` - called when the parser starts to evaluate `rule`
+ - `DBG_MATCH` - called when `rule` is matched, at which point buffer holds entire matched string
+ - `DBG_NOMATCH` - called when the parsers determines that the input does not match currently evaluated `rule`
+
+Very simple implementation could look like this:
+
+```C
+static const char* dbg_str[] = {"Evaluating rule", "Matched rule", "Abandoning rule"};
```

Comment:
I like `static const char *dbg_str` rather than `static const char* dbg_str`.
I use this manner in PackCC consistently.

Reply:
Do you use any automatic formatter (like clang-format, uncrustify, astyle, ...)? If yes, it should be rather easy to incorporate it into tests, so anyone will know that the committed code is correctly formatted as soon as the checks in PR finish. Or even before push, if they run the tests locally.

Reply:
No, I don't use any automatic formatter.
I'll research their features in detail. Thank you for the information.

Reply:
I'd suggest `clang-format`, if you don't use any yet. It is easy to obtain (it is usually installed with clang package), multiplatform, configurable and fast.

I can prepare config file based on the style you use, just give me couple days.

Reply:
I'm very glad if you can. Thanks a lot.